pallets/fragments/src/lib.rs

```diff
@@ -1383,45 +1384,63 @@ pub mod pallet {
 		/// # Arguments
 		///
 		/// * `origin` - The origin of the extrinsic function
-		/// * `definition_hash` - **ID of the Fragment Definition** to **detach**
+		/// * `definition_hash` - **ID of the Fragment Definition**
+		/// * `edition_ids` - **Edition IDs** of the **Fragment Instances that will be detached**
 		/// * `target_chain` - **External Blockchain** to attach the Proto-Fragment into
 		/// * `target_account` - **Public Account Address in the External Blockchain `target_chain`**
 		///   to assign ownership of the Proto-Fragment to
 		#[pallet::weight(25_000)] // TODO - weight
 		pub fn detach(
 			origin: OriginFor<T>,
 			definition_hash: Hash128,
-			edition_id: InstanceUnit,
-			copy_id: InstanceUnit,
+			edition_ids: Vec<InstanceUnit>,
 			target_chain: SupportedChains,
 			target_account: BoundedVec<u8, T::DetachAccountLimit>, // an eth address or so
 		) -> DispatchResult {
-
 			let who = ensure_signed(origin)?;
 
+			ensure!(!edition_ids.is_empty(), Error::<T>::InstancesToDetachIsEmpty);
 			ensure!(!target_account.is_empty(), Error::<T>::DetachAccountIsEmpty);
 
-			let current_block_number = <frame_system::Pallet<T>>::block_number();
+			// let current_block_number = <frame_system::Pallet<T>>::block_number();
+
+			let definition_permissions =
+				<Definitions<T>>::get(definition_hash).ok_or(Error::<T>::NotFound)?.permissions;
 
-			// owner must own instance
-			let owned_instances = <Inventory<T>>::get(who, definition_hash).ok_or(Error::<T>::NoPermission)?;
 			ensure!(
-				owned_instances.contains(&(Compact(edition_id), Compact(copy_id))),
+				(definition_permissions & FragmentPerms::COPY) != FragmentPerms::COPY,
 				Error::<T>::NoPermission
 			);
 
-			// REVIEW - no go if will expire this block
-			let instance = <Fragments<T>>::get((definition_hash, edition_id, copy_id))
-				.ok_or(Error::<T>::NotFound)?;
-			if let Some(expiring_at) = instance.expiring_at {
```

Comment:
I think for now any item that can expire should not be detached at all

Reply:
But if a Fragment Definition is not copyable, it cannot expire. That's why I don't check